core/src/main/scala/org/apache/spark/MapOutputTracker.scala

```diff
@@ -305,7 +305,7 @@ private[spark] class MapOutputTrackerMaster(conf: SparkConf)
 
     if (mapStatuses.contains(shuffleId)) {
       val statuses = mapStatuses(shuffleId)
-      if (statuses.nonEmpty) {
+      if (statuses.nonEmpty && statuses.exists(_ != null)) {
```

Comment:
These two correctness occurs while I revert my modify back to verify my test case will fails. but when I do this, it always encounter some strange problem. 
while we submitMissingTask, we will call `getPreferredLocs` for each task. and in that, for rdd's each ShuffleDependency, it will call `mapOutputTracker.getLocationsWithLargestOutputs(`, and in that, it will judge status.nonEmpty(), so it alway true,
 so it will always call `val status = statuses(mapIdx); val blockSize = status.getSizeForBlock(reducerId)`